libs/langchain/src/agents/middlewareAgent/middleware/bigTool.ts

```diff
@@ -0,0 +1,239 @@
+import { z } from "zod/v3";
+import { BaseChatModel } from "@langchain/core/language_models/chat_models";
+import type { InferInteropZodInput } from "@langchain/core/utils/types";
+import { BaseMessage, HumanMessage } from "@langchain/core/messages";
+
+import { createMiddleware } from "../middleware.js";
+import { initChatModel } from "../../../chat_models/universal.js";
+import type { ModelRequest } from "../types.js";
+
+/**
+ * Zod schema for tool selection structured output.
+ */
+const ToolSelectionSchema = z.object({
+  selectedTools: z.array(z.string()).describe("List of selected tool names"),
```

Comment:
Could we generate this dynamically so it's not a list of string, but a list of possible enums with descriptions?